jcl/source/common/JclSysInfo.pas

```diff
@@ -4293,6 +4294,29 @@ function GetWindowsReleaseName: String;
     Result := '';
 end;
 
+function GetWindowsReleaseCode: String;
+begin
+  // Looks much like the 'GetWindowsReleaseCodeName', except for the Windows 10 versions
+  // prior to Release Id 1903 - those have a different 'code' vs the 'code name'.
+  if IsWin10 then
+  begin
+    case GetWindowsReleaseId of
+      1903:
+        Result := '19H1';
+      1909:
+        Result := '19H2';
+      2004:
+        Result := '20H1';
+      2009:
+        Result := GetWindowsDisplayVersion;
+      else
+        Result := IntToStr(GetWindowsReleaseId);
```

Comment:
Above comment suggests versions _prior_ to 1903 should be treated differently, but having this code in the `else` statement block of a `case` statement, will also be hit in future versions. Therefor, if this code really has to do what the comment says, an `if GetWindowsReleaseId  < 1903 then Result := Result := IntToStr(GetWindowsReleaseId) else` block would have to precede this `case` statement

Reply:
Hmmm, you're right @PatrickvL - I'll refactor it

Reply:
Changed code